Feature/add item and model

alexlarin12 opened the pull request:
1.  в GarageCollection - реализовал выборочное удаление ячейки с автомобилем (предварительно появляется AlertController)
2. создал экран с UIPickerView для выбора марки и модели авто - данные получаем из JSON файла в ресурсах
3. реализовал отображение выбранной марки и модели сразу под текстфилдом
4. реализовал побуквенную фильтрацию марок авто при заполнении тексфилда с последующей передачей данных при нажатии кнопки Назад

alexlarin12 committed:
рефакторинг GarageController

добавил выборочное удаление авто из ячеек GarageCollectionView

alexlarin12 committed:
UIPickerView

создал uipicker

alexlarin12 committed:
парсер json

alexlarin12 committed:
добавление марки и модели

1. файл json со всеми авто положил в ресурсы
2. создал PickerView с обработкой данных словаря
3. декодирование json в классе CarDecodable и прописал методы для работы со словарем
4. переделал свойства текстфилдов в ячейках - при нажатии на марку, год, тип топлива - текстфилд не редактируется

alexlarin12 committed:
UIPicker

1. добавил поиск через текстфилд при добавлении букв
2. Убрал баг при котором  марка и модель в самой первой ячейке массива - сохраняется

aksilont commented on MyCar/Environments/Info.plist:
```diff
@@ -2,8 +2,6 @@
 <plist version="1.0">
 <dict>
-	<key>UIUserInterfaceStyle</key>
-	<string>Dark</string>
```
Специально отключил темную тему?

alexlarin12 replied:
не отключал. Как то видимо само получилось

aksilont commented on MyCar/App/Flows/Scenes/Garage/GarageCollectionViewCell.swift:
```diff
@@ -46,12 +46,20 @@ class GarageCollectionViewCell: UICollectionViewCell {
         let checkMark = SSCheckMark()
-        // checkMark.backgroundColor = #colorLiteral(red: 0.6666666865, green: 0.6666666865, blue: 0.6666666865, alpha: 1)
-        checkMark.backgroundColor = UIColor(red: 0.6666666865, green: 0.6666666865, blue: 0.6666666865, alpha: 1)
+        checkMark.backgroundColor = UIColor(red: 170/255, green: 170/255, blue: 170/255, alpha: 1)
```
Цвет надо бы в extension перенести

aksilont commented on MyCar/App/Flows/Scenes/Garage/GarageCollectionViewCell.swift:
```diff
@@ -92,14 +101,21 @@ class GarageCollectionViewCell: UICollectionViewCell {
+            deleteButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: 0),
+            deleteButton.centerYAnchor.constraint(equalTo: checkMarkView.centerYAnchor),
+            deleteButton.widthAnchor.constraint(equalToConstant: 30),
+            deleteButton.heightAnchor.constraint(equalToConstant: 30) ])
```
Магические цифры :)

aksilont commented on MyCar/App/Flows/Scenes/AddItemAndModel/AddItemAndModelViewController.swift:
```diff
@@ -0,0 +1,178 @@
+            autoLabel.topAnchor.constraint(equalTo: infoTextField.bottomAnchor, constant: 10),
+            autoLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            autoLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            carPicker.topAnchor.constraint(equalTo: infoTextField.bottomAnchor, constant: 10),
```
Как нам на уроках говорили магические цифры :)
Рекомендую создать переменные с этими константами

alexlarin12 replied:
Просто обЪявить в классе let topValue = 10, let leadingValue = 20? и т.д.??

aksilont replied:
Ага, что-то вроде этого, leadingInsets и т.д.

aksilont commented on MyCar/App/Flows/Scenes/AddAuto/AddAutoViewController.swift:
```diff
@@ -30,7 +30,7 @@ class AddAutoViewController: UIViewController {
         let button = UIButton()
         button.sizeToFit()
-        button.backgroundColor = #colorLiteral(red: 0.2172074616, green: 0.6069719195, blue: 0.6331881881, alpha: 1)
+        button.backgroundColor = UIColor(red: 54/255, green: 155/255, blue: 161/255, alpha: 1)
```
Цвет надо бы в extension перенести

alexlarin12 replied:
с цветом понял.

aksilont commented on MyCar/App/Flows/Scenes/Garage/CheckMark.swift:
```diff
@@ -103,7 +103,7 @@ class SSCheckMark: UIControl {
         bezierPath.addLine(to: CGPoint(x: group.minX + 0.75000 * group.width, y: group.minY + 0.35417 * group.height))
         bezierPath.lineCapStyle = CGLineCap.square
-        #colorLiteral(red: 0.6666666865, green: 0.6666666865, blue: 0.6666666865, alpha: 1).setStroke()
+        UIColor(red: 170/255, green: 170/255, blue: 170/255, alpha: 1).setStroke()
```
Цвет надо бы в extension перенести

pkslty commented:
У меня общие замечания по интерфейсу и логике работы
1. При выборе модели не очень очевидно, какая модель будет выбрана оттого, что в строке поиска жирный и более заметный шрифт. При открытии экрана первая по списку из всех модель уже вроде бы выбрана, но нет.
2. Поиск модели не работает, только марка. А для модели приходится крутить барабан пикера.
3. При нажатии на марку и модель в автолэйбле происходит что-то неочевидное. Отчитывается назад на несколько позиций.
4. А что если в списке не окажется нужной машины, никак не решаемая проблема?

alexlarin12 committed:
refactoring по результатам полезной критики)

1. цвета убрал в extension
2. величины основных констрэинт сделал ввиде переменных
3. В выборе авто с пикером - для лейбла выбранного авто - увеличил шрифт, для текстфилда - сделал шрифт light
4. исправил баг - первая по списку модель передается в AddAuto
5. Если в списке не окажется модели - сделал ручное добавление/коррекция в AddAuto
6. Подчистил CoreData repository

aksilont committed:
.DS_store deleted
